quantum/action.c

```diff
@@ -970,9 +970,7 @@ __attribute__((weak)) void unregister_code(uint8_t code) {
  */
 __attribute__((weak)) void tap_code_delay(uint8_t code, uint16_t delay) {
     register_code(code);
-    for (uint16_t i = delay; i > 0; i--) {
-        wait_ms(1);
-    }
+    wait_ms(delay);
```

Comment:
I believe that this needs to be reverted, as wait_ms may need an actual set amount at compile time on some platforms. Hence the for loop here.

Reply:
That should have been fixed for AVR long ago (and other platforms never had that problem):
```c
#define wait_ms(ms)                             \
    do {                                        \
        if (__builtin_constant_p(ms)) {         \
            _delay_ms(ms);                      \
        } else {                                \
            for (uint16_t i = ms; i > 0; i--) { \
                _delay_ms(1);                   \
            }                                   \
        }                                       \
    } while (0)
```